src/auth/authApp.js

```diff
@@ -405,51 +494,44 @@ export function handlePendingSignIn(nameLookupURL: string = '',
           .then((response) => {
             if (!response.ok) { // return blank profile if we fail to fetch
               userData.profile = Object.assign({}, DEFAULT_PROFILE)
-              window.localStorage.setItem(
-                BLOCKSTACK_STORAGE_LABEL, JSON.stringify(userData)
-              )
+              const sessionData = caller.store.getSessionData()
+              sessionData.userData = userData
+              caller.store.setSessionData(sessionData)
               return userData
             } else {
               return response.text()
                 .then(responseText => JSON.parse(responseText))
                 .then(wrappedProfile => extractProfile(wrappedProfile[0].token))
                 .then((profile) => {
+                  const sessionData = caller.store.getSessionData()
                   userData.profile = profile
-                  window.localStorage.setItem(
-                    BLOCKSTACK_STORAGE_LABEL, JSON.stringify(userData)
-                  )
+                  sessionData.userData = userData
+                  caller.store.setSessionData(sessionData)
                   return userData
                 })
             }
           })
       } else {
+        const sessionData = caller.store.getSessionData()
         userData.profile = tokenPayload.profile
-        window.localStorage.setItem(
-          BLOCKSTACK_STORAGE_LABEL, JSON.stringify(userData)
-        )
+        sessionData.userData = userData
+        caller.store.setSessionData(sessionData)
         return userData
       }
     })
 }
 
 /**
  * Retrieves the user data object. The user's profile is stored in the key `profile`.
- * @return {Object} User data object.
- */
-export function loadUserData() {
-  return JSON.parse(window.localStorage.getItem(BLOCKSTACK_STORAGE_LABEL))
-}
-
-/**
- * Sign the user out and optionally redirect to given location.
- * @param  {String} [redirectURL=null] Location to redirect user to after sign out.
- * @return {void}
+ *
+ *  @param {UserSession} caller - the instance calling this function
+ *  @return {Object} User data object.
+ *  @private
  */
-export function signUserOut(redirectURL: ?string = null) {
-  window.localStorage.removeItem(BLOCKSTACK_STORAGE_LABEL)
-  window.localStorage.removeItem(BLOCKSTACK_GAIA_HUB_LABEL)
-
-  if (redirectURL !== null) {
-    window.location = redirectURL
+export function loadUserDataImpl(caller: UserSession) {
+  const userData = caller.store.getSessionData().userData
+  if (!userData) {
```

Comment:
This is a breaking change in expected functionality. The previous `loadUserData` implementation returns `null` if the user is signed out, but now we are throwing an error.

Reply:
Although it's changing the behaviour, I would say that throwing an error would be more correct in this case. Can we just make a note of this change in the change logs?

Reply:
Well we already did the work to make the upgrade easier for folks who want to use v19 without a `UserSession`. Ultimately we want to encourage adoption of this as much as possible. With your backwards compatibility changes, you've made it so that almost every developer can upgrade without changing their code. This would change that. I know I've personally a lot of written code that checks for `loadUserData()` without first calling `isUserSignedIn()`

Reply:
I would argue that the point of the backward compatibility changes were to make it easier to migrate to v19. This change in behaviour is more of a bug fix. I think it should be okay as long as we make it clear that there will be some changes required to upgrade. This is after all a major new version.

Reply:
Yeah that was my point, we want to make it easier to upgrade. This would actually be the hardest part of upgrading, because so many people use it. It's not like we've documented that it's supposed to throw an error.

Reply:
But, you're the maintainer, and I will happily defer to what you think is best!

Reply:
Let's keep it this way for now, if it turns out to be a big problem during the beta we can change it back.